### ISSUE TITLE
Catmul Rom Splines amd Bezier Curves 

### ISSUE THREAD

**Issue by loopyd** [OneLoneCoder/olcUTIL_Geometry2D#63]
The important question to ask over one of javidx9's [older videos](https://youtu.be/9_aJGUTePYo?si=7ytF_iBdWbC6PsFM) is should catmul roms and bezier curves become a part of geometry2d lib?

The work has already been done to optimize and solve the linear regression crisis, smooth interpolation for constant velocity and intersection.  

My personal take:

I feel that they do from geo2ds point and line primitives deserve their own types but this feature warrants feedback from community before going out and bringing it in. 

As a long time watcher of Javid's channel have got to strongly recommend in favor of these primitive types.   Curves are everywhere and extremely useful for 2D games.

**Comment by Piratux** [comment]
I support adding splines, since most implemented function checks (such as intersects, overlaps, etc.) make sense.

**Comment by OneLoneCoder** [comment]
Agreed, I'll add some sort of placeholder for spline - we'll need some real maths experts though to start integrating them into operations with the other shapes. That said, having a robust spline primitive is useful in its own right. I'll add the basics for now.
